dspace-api/src/main/java/org/dspace/content/DSpaceObjectServiceImpl.java

```diff
@@ -234,12 +239,19 @@ public void addMetadata(Context context, T dso, MetadataField metadataField, Str
                             List<String> authorities, List<Integer> confidences) throws SQLException {
         boolean authorityControlled = metadataAuthorityService.isAuthorityControlled(metadataField);
         boolean authorityRequired = metadataAuthorityService.isAuthorityRequired(metadataField);
-
         // We will not verify that they are valid entries in the registry
         // until update() is called.
         for (int i = 0; i < values.size(); i++) {
 
+            if (authorities != null && authorities.size() >= i) {
+                if (StringUtils.startsWith(authorities.get(i), Constants.VIRTUAL_AUTHORITY_PREFIX)) {
```

Comment:
The design model was discussed and decided a while ago.

Reply:
the design was discussed but this and the enormous amount of code needed so far is imho a clear sign that the full implication were not clear or at least not clearly presented.
I still see only the complexity cost of the current solution for the relationship management and no benefit, before to close this conversation I would like to see an answer (which are the proven and the expected benefits?). Otherwise, feel free to keep it as an unresolved conversation about the current disagreement about the design choice